onnxruntime/core/providers/coreml/builders/impl/gemm_op_builder.cc

```diff
@@ -146,6 +146,7 @@ bool GemmOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderInputPara
       return false;
     }
 
+    // TODO is it ok if the shape is dynamic and empty?
```

Comment:
note: disabling dynamic shapes with zero elements for now by erroring out at runtime